lib/filewatch/discoverer.rb

```diff
@@ -35,7 +35,7 @@ def discover
 
     def can_exclude?(watched_file, new_discovery)
       @exclude.each do |pattern|
-        if watched_file.pathname.fnmatch?(pattern)
+        if watched_file.pathname.basename.fnmatch?(pattern)
```

Comment:
the [WatchedFile class already provides a `filename` accessor for the basename](https://github.com/logstash-plugins/logstash-input-file/blob/master/lib/filewatch/watched_file.rb#L20), so we can replace instead with:

```suggestion
        if watched_file.filename.fnmatch?(pattern)
```

Reply:
This is not correct. The `filename` method returns a String, not a PathName, so `fnmatch?` cannot be used.

```
  Error: undefined method `fnmatch?' for "ignoreme":String
  Exception: NoMethodError
  Stack: /home/user/elk/logstash-7.5.1/vendor/bundle/jruby/2.5.0/gems/logstash-input-file-4.1.11/lib/filewatch/discoverer.rb:38:in `block in can_exclude?'
org/jruby/RubyArray.java:1800:in `each'
/home/user/elk/logstash-7.5.1/vendor/bundle/jruby/2.5.0/gems/logstash-input-file-4.1.11/lib/filewatch/discoverer.rb:37:in `can_exclude?'
```

Reply:
For the record, I tested using this configuration (in `test.conf`):
```
input {
    file {
        start_position => beginning
        path => ["/home/user/elk/testlog/*"]
        exclude => ["ignoreme"]
    }
}
```

And this command:
`rm -rf logstash-7.5.1/data/plugins/; logstash-7.5.1/bin/logstash -f test.conf --log.level=trace 2>&1 | tee logstash.log`

Then examining the output. The `tee` command saves the log to the `logstash.log` file so it can be searched using one's preferred text editor.

Reply:
my brain is still in 2019, LGTM